File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
import math
def solution(number, limit, power):
    full_attack =0
    
    for i in range(1,number+1):
        attack = 0    
        sqrt_i=int(math.sqrt(i))
        for j in range(1,sqrt_i+1):
            if i%j ==0:
                attack+=2;
                if j*j==i:
                    attack-=1;
                
        if attack >limit:
            attack = power;
        full_attack += attack
    
    return full_attack

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let full_attack = 0;
    for (let i= 1;i<=number;i++){
        let attack = 0; // attack : 공격력값
        for(let j =1;j*j<=i;j++){
            if (i % j == 0){
                attack+=2; // 약수라면 공격력1을 추가,
                if (j*j ==i){
                    attack-=1;
                }
            }
        }
        if (attack >limit){ // 공격력이 limit보다 크면 공격력은 power로 치환
            attack = power 
    
            }
    full_attack += attack;
    }
    
    return full_attack;
}